### PR TITLE
Makes ExDoc generate Valid HTML5 pages

### PR DESCRIPTION
**Pull request by eksperimental**
This commits fixes every single bug,
thus it makes the docs generated (checked only on the Elixir docs)
valid HTML5 documents.

I throughly tested everything,
we didn't have that many issues. but they were hard to fix,
I have sent a few a few PRs to earmark because some issues came from there.
there is on last PR that needs to be reviewed and merged.

I have validated the Elixir docs using html-proofer and html5validator,
no internal broken links found, only one issue and we are 100% valid.
I will may need help to fix it.

OK. trying to push this so we can test the longer the better..

### CODE REVIEW

File: lib/ex_doc/formatter/html/templates/sidebar_template.eex [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
   <div class="sidebar-search">
     <i class="icon-search"></i>
-    <input type="text" class="sidebar-searchInput" placeholder="search" autocomplete="off" results="0" />
+    <input type="text" class="sidebar-searchInput" placeholder="search" autocomplete="off" />
```
Comment (eksperimental) [inline lib/ex_doc/formatter/html/templates/sidebar_template.eex]: @Dignifiedquire I have removed "results" was it was not a valid attribute, I think it's webkit specific. Could you please add this via Jquery? So we don't need to worry about it when we run the validators? let me know if that's still ok.

Comment (Dignifiedquire) [inline lib/ex_doc/formatter/html/templates/sidebar_template.eex]: I don't think we use it anymore, so thanks for cleaning up :)

**Comment by Dignifiedquire** [comment]
Thanks :octocat: lgtm

File: lib/ex_doc/formatter/html/templates/sidebar_template.eex (hunk 1 above)
Comment (eksperimental) [inline lib/ex_doc/formatter/html/templates/sidebar_template.eex]: even better then

**Pull request merged**